src/chains/algorand_1/algoTransaction.ts

```diff
@@ -267,25 +299,51 @@ export class AlgorandTransaction implements Transaction {
     }
   }
 
-  /** Returns address, for which, a matching signature must be attached to transaction
-   *  Throws if actions[].from is not a valid address - needed to determine the required signature */
+  /** Returns address, for which, a matching signature must be attached to transaction */
   public get missingSignatures(): AlgorandAddress[] {
     this.assertIsValidated()
-    const missingSignature = this.hasAllRequiredSignatures ? null : this.requiredAuthorization
-    return isNullOrEmpty(missingSignature) ? null : [missingSignature] // if no values, return null instead of empty array
+    const missingSignatures = this.requiredAuthorizations?.filter(
```

Comment:
if requiredAuthorizations is null, this will return null. Add a || [] on end of the filter block

Reply:
done

Reply:
done